wallet/src/actors/app/methods.rs

```diff
@@ -634,7 +636,17 @@ impl App {
             self.state
                 .get_wallet_by_session_and_id(&session_id, &wallet_id),
         )
-        .and_then(move |_wallet, slf: &mut Self, _| slf.send_inventory_transaction(transaction));
+        .and_then(move |wallet, slf: &mut Self, _| {
+            slf.send_inventory_transaction(transaction.clone())
+                .and_then(move |value, _slf, _ctx| {
+                    let _ = wallet.add_local_movement(&model::ExtendedTransaction {
+                        transaction,
+                        metadata: None,
+                    });
```

Comment:
This is ignoring any errors returned by `add_local_movement`. Try this to return the errors:
```suggestion
                    if let Some(err) = wallet.add_local_movement(&model::ExtendedTransaction {
                        transaction,
                        metadata: None,
                    }) {
                        return actix::fut::err(err);
                    }
```